### PR TITLE
Update LICENSE file

### PR DESCRIPTION
**Pull request by tamiko**
As discussed, this commit unifies the information contained in our previous
`LICENSE` file and the one found on the webpage
https://www.dealii.org/license.html

I have improved the wording a bit and removed unnecessary information (so that
everything is in one place now). The most important change is the switch to DCO
1.1. We are basically stating thateverything is in one place now). The most
important change is the switch to DCO 1.1, which is a much better fit for a
repository that contains multiple licenses.

### CODE REVIEW

**Comment by tamiko** [comment]
Preview: https://github.com/tamiko/dealii/blob/update_legalese/LICENSE.md

Justification: The deal.II codebase is nowadays forked and redistributed all over github. We should really make sure that the repository contains such an explicit anchor point (that conveniently points back to the authors and publication list as well.)

File: LICENSE.md [hunk 1]
```diff
@@ -1,25 +1,127 @@
...
+
+This is the usual, fair way of giving credit to contributors to a
+scientific result. In addition, it helps us justify our effort in
+developing deal.II as an academic undertaking.
+
+We keep a list of publications using deal.II. Feel free to let us know
+about your publications.
+
+
+Parts of the deal.II distribution under a separate license
```
Comment (jppelteret) [inline LICENSE.md]: distribution -> distributed

File: LICENSE.md [hunk 2]
```diff
@@ -1,25 +1,127 @@
...
+    of deal.II should state the name of the library and cite appropriate
+    deal.II references listed at the top of the [publications
+    page](https://www.dealii.org/publications.html).
+
+This is the usual, fair way of giving credit to contributors to a
+scientific result. In addition, it helps us justify our effort in
+developing deal.II as an academic undertaking.
+
+We keep a list of publications using deal.II. Feel free to let us know
+about your publications.
```
Comment (jppelteret) [inline LICENSE.md]: ...about your publications so that we can add them to the aforementioned list.

File: LICENSE.md [hunk 3]
```diff
@@ -1,25 +1,127 @@
...
+
+
+Parts of the deal.II distribution under a separate license
+----------------------------------------------------------
+
+The subdirectory `bundled/` contains third party software. <b>Please note
+that the software there is copyrighted by others than the deal.II
```
Comment (jppelteret) [inline LICENSE.md]: software located there

Comment (jppelteret) [inline LICENSE.md]: by others than the deal.II authors -> by their respective authors (independent to the deal.II authors)

File: LICENSE.md [hunk 4]
```diff
@@ -1,25 +1,127 @@
...
+authors</b> and are covered by different licenses.
+
+The libraries listed above are all open source and as such place few
+restrictions on their use. However, if you want to distribute software
+developed with deal.II (in source or binary form) and you are using the
+packages above (with source code in `bundled/`), then they may impose
+different terms. Please consult the licenses of these packages for more
+information.
```
Comment (jppelteret) [inline LICENSE.md]: Perhaps we should drop a courtesy note here suggesting again that users cite the appropriate documentation for these libraries that they use?

File: LICENSE.md (hunk 1 above)
Comment (bangerth) [reply LICENSE.md]: Or better "Parts of deal.II are distributed".  Alternatively, "Parts of the deal.II distributed are under..."

File: LICENSE.md [hunk 5]
```diff
@@ -1,25 +1,128 @@
...
+    of deal.II should state the name of the library and cite appropriate
+    deal.II references listed at the top of the [publications
+    page](https://www.dealii.org/publications.html).
+
+This is the usual, fair way of giving credit to contributors to a
+scientific result. In addition, it helps us justify our effort in
+developing deal.II as an academic undertaking.
+
+We keep a list of publications using deal.II. Feel free to let us know
+about your publications so that we can add them to the aforementioned list.
```
Comment (bangerth) [inline LICENSE.md]: Can you reword this to say "Please let us know about your publications so .... list, by emailing the reference information to one of the Principal Developers of deal.II" (and add a link to the authors page)?

File: LICENSE.md (hunk 1 above)
Comment (tamiko) [reply LICENSE.md]: I already changed this to "Bundled third party software in the deal.II repository". I'd prefer to keep it that way.

File: LICENSE.md (hunk 5 above)
Comment (tamiko) [reply LICENSE.md]: Jup.

File: LICENSE.md (hunk 1 above)
Comment (bangerth) [reply LICENSE.md]: That works.

File: LICENSE.md (hunk 5 above)
Comment (bangerth) [reply LICENSE.md]: I meant to get rid of the "Feel free to..." part. I don't want people to feel free, I want them to do it!

Comment (tamiko) [reply LICENSE.md]: :roll_eyes:

**Comment by bangerth** [comment]
Yes, we seem to have a habit doing so. The fact that there is a word for that sort of implies that others do that too.
